backend/sites/src/app/services/folio/folioContents.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FolioContents } from '../../entities/folioContents.entity';
+import { Repository } from 'typeorm';
+import { FolioContentDTO } from '../../dto/folioContent';
+import { plainToInstance } from 'class-transformer';
+
+@Injectable()
+export class FolioContentsService {
+  constructor(
+    @InjectRepository(FolioContents)
+    private folioContentRepository: Repository<FolioContents>,
+  ) {}
+
+
+  async getSiteForFolio(folioId: string): Promise<FolioContents[]> {
+    try {
+      const sitesForFolioArr = await this.folioContentRepository.find({  
+        relations: ['site','folio'],
+        where: { folioId },
+      });
+      return sitesForFolioArr;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+
+  async addFolioContent(inputDTO: FolioContentDTO): Promise<boolean> {
+    const { siteId, folioId } = inputDTO;
+    try {
+      const existingRecord = await this.folioContentRepository.findOne({
+        where: { siteId, folioId },
+      });
+
+      if (existingRecord) {
+        return false;
+      } else {
+        delete inputDTO.id;
```

Comment:
Why is this deletion required, if the record is not existing we can straight add it, isn't?

Reply:
i added it because I wanted to delete the property before passing to plainToInstance. anyway I have update the code and no longer using it.